community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/StandalonePageCacheFactory.java

```diff
@@ -34,22 +34,27 @@
  */
 public final class StandalonePageCacheFactory
 {
-
     private StandalonePageCacheFactory()
     {
     }
 
     public static PageCache createPageCache( FileSystemAbstraction fileSystem )
     {
-        return createPageCache( fileSystem, PageCacheTracer.NULL );
+        return createPageCache( fileSystem, null, PageCacheTracer.NULL );
+    }
+
+    public static PageCache createPageCache( FileSystemAbstraction fileSystem, Integer pageSize )
+    {
+        return createPageCache( fileSystem, pageSize, PageCacheTracer.NULL );
     }
 
-    public static PageCache createPageCache( FileSystemAbstraction fileSystem, PageCacheTracer tracer )
+    public static PageCache createPageCache( FileSystemAbstraction fileSystem, Integer pageSize,
```

Comment:
I agree with this sentiment. You could add another static constructor that takes a `PageSwapperFactory` and a page size, and then move the construction of the default page swapper factory up one level in the stack. The creation & configuration of this page swapper factory can also be extracted into a static helper method. That way you would not need any null-logic at all.